x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearningUsageTransportAction.java

```diff
@@ -227,6 +268,66 @@ private void addDataFrameAnalyticsUsage(GetDataFrameAnalyticsStatsAction.Respons
         }
     }
 
+    private static void initializeStats(Map<String, Long> emptyStatsMap) {
+        emptyStatsMap.put("sum", 0L);
+        emptyStatsMap.put("min", 0L);
+        emptyStatsMap.put("max", 0L);
+    }
+
+    private static void updateStats(Map<String, Long> statsMap, Long value) {
+        statsMap.compute("sum", (k, v) -> v + value);
+        statsMap.compute("min", (k, v) -> Math.min(v, value));
+        statsMap.compute("max", (k, v) -> Math.max(v, value));
+    }
+
+    //TODO separate out ours and users models possibly regression vs classification
+    private void addTrainedModelStats(SearchResponse response, Map<String, Object> inferenceUsage) {
+        inferenceUsage.put("trained_models",
+            Collections.singletonMap(MachineLearningFeatureSetUsage.ALL, createCountUsageEntry(response.getHits().getTotalHits().value)));
+    }
+
+    //TODO separate out ours and users models possibly regression vs classification
+    private void addInferenceIngestUsage(NodesStatsResponse response, Map<String, Object> inferenceUsage) {
+        Set<String> pipelines = new HashSet<>();
+        Map<String, Long> docCountStats = new HashMap<>(3);
+        Map<String, Long> timeStats = new HashMap<>(3);
+        Map<String, Long> failureStats = new HashMap<>(3);
+        initializeStats(docCountStats);
+        initializeStats(timeStats);
+        initializeStats(failureStats);
+
+        response.getNodes()
+            .stream()
+            .map(NodeStats::getIngestStats)
+            .map(IngestStats::getProcessorStats)
+            .forEach(map ->
+                map.forEach((pipelineId, processors) -> {
+                    boolean containsInference = false;
+                    for(IngestStats.ProcessorStat stats : processors) {
+                        if (stats.getName().equals(InferenceProcessor.TYPE)) {
+                            containsInference = true;
+                            long ingestCount = stats.getStats().getIngestCount();
+                            long ingestTime = stats.getStats().getIngestTimeInMillis();
+                            long failureCount = stats.getStats().getIngestFailedCount();
+                            updateStats(docCountStats, ingestCount);
+                            updateStats(timeStats, ingestTime);
+                            updateStats(failureStats, failureCount);
+                        }
+                    }
+                    if (containsInference) {
+                        pipelines.add(pipelineId);
+                    }
+                })
+            );
+
+        Map<String, Object> ingestUsage = new HashMap<>(4);
+        ingestUsage.put("pipelines", createCountUsageEntry(pipelines.size()));
+        ingestUsage.put("docs", docCountStats);
```

Comment:
Am I right that this is going to create an object called `docs` containing the fields `min`, `max` and `sum`?  It would be unclear to me what that meant.  Is it `docs_processed_per_ingest_node`?  That's probably too verbose, but I think the 3 object names need to convey that the values are per node.  Maybe we should discuss in a call.

Reply:
Possibly `docs_processed`. This is not `per_node`. It is a summation across all nodes and only contains data pertaining to the `inference` processor.